File: backend/app/schemas/audit.py
"""Audit log response schemas."""
from datetime import datetime
from typing import Any, Dict, List, Optional
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class AuditLogEntry(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    user_id: Optional[UUID] = None
    user_email: Optional[str] = None
    action: str
    resource_type: Optional[str] = None
    resource_id: Optional[str] = None
    ip_address: Optional[str] = None
    details: Optional[Dict[str, Any]] = None
    created_at: datetime


class AuditLogListResponse(BaseModel):
    total: int
    page: int
    limit: int
    items: List[AuditLogEntry]

File: frontend/src/api/audit.js
import { api } from './client';

export const audit = {
  list: (page = 1, limit = 50, action = '') => {
    let qs = `?page=${page}&limit=${limit}`;
    if (action) qs += `&action=${encodeURIComponent(action)}`;
    return api.get(`/audit${qs}`);
  },
};
